### ISSUE TITLE
Two peers sending video/audio leads to immediate renegotiation

### ISSUE THREAD

**Issue by feross** [feross/simple-peer#670]
Using `simple-peer` in a relatively simple setup. Seeing the non-initiator peer immediately request renegotiation even though everything is specified upfront in the constructor. I see two offer/answers sent from the initiator peer.

Here's my setup: Two peers, each initialized with the `stream` option, so `addTrack()` is called twice (once for video, once for audio) internally in the constructor. For the non-initiator peer, `this._isNegotiating` is already `true` because of this line in the constructor:

https://github.com/feross/simple-peer/blob/90c3dfd35ef3727b17e7b52b1c45cbc84d6ff416/index.js#L89

Therefore, for the non-initiator peer, the calls to `this._needsNegotiation()` in response to `addTrack()` cause a negotiation to be immediately queued. Once the connection is stable it triggers the re-negotiation. 

Is this intentional?

**Comment by feross** [comment]
I have a draft PR that fixes the problem, but in a hacky way: https://github.com/feross/simple-peer/pull/671

Thoughts?

**Comment by nazar-pc** [comment]
I agree, it seems to be a bit wrong. I'd expect renegotiation to not be triggered before the first offer was received and answer for it was generated. It is just pointless in this case.

**Comment by t-mullen** [comment]
I've implemented a fix for this in #694, with tests and more debugging logs.

The causes were multiple:
- #570 changed the initial value of `_isNegotiating` to use the `channelConfig.negotiated` option (whether the datachannel is negotiated isn't relevant to the negotiation of the peer).
- `_firstStable` was set in the wrong place.
- The entire approach of dropping the non-initiator's 1st queued renegotiation request on first stable is wrong - this may be a valid request if we add a track right after creating an answer. What we actually want to do is start negotiation on both peers and immediately discard the non-initiator's 1st batched negotiation (since we know the initiator will negotiate). This was an unlikely race condition, but I've fixed it anyways.
